client/pages/post-form.jsx

```diff
@@ -13,8 +15,9 @@ export default class PostForm extends React.Component {
 
   handleSubmit(event) {
     event.preventDefault();
+    const userId = this.context.user.userId;
     const formData = new FormData(event.target);
-    fetch('/api/post-form', {
+    fetch(`/api/post-form/user/${userId}`, {
```

Comment:
As mentioned before, these should all include the token. The server needs to use the token to identify the user. The client is not allowed to claim "i am user id 1". They might be a liar.